### PR TITLE
Presidio power skill

### PR DESCRIPTION
**Pull request by shiranr**
## Purpose
Add new Power Skill to the repository. PII anonymization using Presidio.

## Does this introduce a breaking change?
```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?
```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone https://github.com/shiranr/azure-search-power-skills.git
cd azure-search-power-skills.git
git checkout presidio_power_skill
npm install
```

* Test the code
The readme contains all the instructions including an updated notebook 

## What to Check
This Power Skill works with PythonFastAPI and built with the same logic as the text summarization. 
The result of the indexing should be anonymized text, all PII entities should be removed from the text.

### CODE REVIEW

File: Text/PresidioPIIAnonymization/README.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+   
+1) ### Run the ACS indexer 
+
+    Populate the values in the [indexer file](deployment/azuresearch/create_indexer.json) or 
+    [Create/Run your indexer](notebooks/PII%20Anonymization.ipynb#Now-we-create-the-indexer)
+
+    The indexer will automatically run and you should see requests coming in if you look at the Web App logs.
+
+1)  ### Test the index 
+    Investigate your indexed data, check the most similar images
```
Comment (bleroy) [inline Text/PresidioPIIAnonymization/README.md:215]: Unclear.

File: Text/PresidioPIIAnonymization/README.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+    Now we are in a position to search on our most similar data, navigate to the [Let's go and test the ACS index](notebooks/PII%20Anonymization.ipynb#Let's-go-and-test-the-ACS-index)
```
Comment (bleroy) [inline Text/PresidioPIIAnonymization/README.md:217]: Add quotes and specify the type of thing we're navigating to.

File: Text/PresidioPIIAnonymization/notebooks/PII Anonymization.ipynb [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "API_KEY = \"[YourSecretKeyCanBeAnything]\"\n", # Set the KEY value you deployed your Web App with [YourSecretKeyCanBeAnything]\n",
+    "URL_LOCAL = f\"http://0.0.0.0:5000/api/extraction\"\n",
```
Comment (bleroy) [inline Text/PresidioPIIAnonymization/notebooks/PII Anonymization.ipynb:33]: There seems to be a formatting issue here, I think a missing quote before the '#'.

**Review by bleroy — CHANGES_REQUESTED** [review]
Thanks for the contributions. Looks good, except for a few things to fix in the readme.

File: Text/PresidioPIIAnonymization/README.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
+# Text PII Anonymization Power Skill
+
+[Presidio](https://github.com/microsoft/presidio) is an open-source tool to recognize, analyze and anonymize personally identifiable information (PII). Using trained ML models, Presidio was built to ensure sensitive text is properly managed and governed.
+
+This Power Skill uses [Presidio](https://github.com/microsoft/presidio) Analyzer and Anonymizer
+to find and remove PII entities. Even though Presidio supports several anonymization methods (hash, encrypt, redact, replace, mask), 
+this Power Skill only uses redact and removes the PIIs completely from the text  .
+
+This skill is ideal for finding and removing PII entities from the search text.
```
[the remark this answers is not in the document]
Comment (shiranr) [reply Text/PresidioPIIAnonymization/README.md:9]: I added it in the next couple of lines.

File: Text/PresidioPIIAnonymization/README.md (hunk 2 above)
Comment (shiranr) [reply Text/PresidioPIIAnonymization/README.md:220]: done.

File: Text/PresidioPIIAnonymization/README.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    ```
+    You will be prompted with:
+    
+    ```bash
+    Do you want to perform these actions?
+      Terraform will perform the actions described above.
+      Only 'yes' will be accepted to approve.
+    ```
+    
+    Type ```bash yes```
```
[the remark this answers is not in the document]
Comment (shiranr) [reply Text/PresidioPIIAnonymization/README.md]: Done

File: Text/PresidioPIIAnonymization/README.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+    [Create the data source](notebooks/PII%20Anonymization.ipynb#Create-the-data-source)
+
+    #### Index
+    Populate your values in the [index file](deployment/azuresearch/create_index.json) or use the 
+    [Create the index](notebooks/PII%20Anonymization.ipynb#Now-we-create-the-index)
+    
+    #### Skillset
+    
+    Populate the values in the [skillset file](deployment/azuresearch/create_skillset.json) or use the 
+    [Create the SkillSet](notebooks/PII%20Anonymization.ipynb#Now-we-create-the-skill-set)
```
[the remark this answers is not in the document]
Comment (shiranr) [reply Text/PresidioPIIAnonymization/README.md]: Done

File: Text/PresidioPIIAnonymization/README.md [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+    instance as we want to augment what we can extract using custom vision with our similarity
+    model.
+    
+    You will need your [ACS API Key](https://docs.microsoft.com/en-us/azure/search/search-security-api-keys)
+    and the URL for your ACS instance. 
+   
+1) ### Run the ACS indexer 
+
+    Populate the values in the [indexer file](deployment/azuresearch/create_indexer.json) or 
+    [Create/Run your indexer](notebooks/PII%20Anonymization.ipynb#Now-we-create-the-indexer)
```
[the remark this answers is not in the document]
Comment (shiranr) [reply Text/PresidioPIIAnonymization/README.md]: Done

File: Text/PresidioPIIAnonymization/README.md (hunk 1 above)
Comment (shiranr) [reply Text/PresidioPIIAnonymization/README.md]: Fixed.

File: Text/PresidioPIIAnonymization/notebooks/PII Anonymization.ipynb (hunk 3 above)
Comment (shiranr) [reply Text/PresidioPIIAnonymization/notebooks/PII Anonymization.ipynb]: fixed.

**Comment by shiranr** [comment]
@bleroy Thanks for the review, I fixed the comments, please LMK if there's anything else to change.

**Review by bleroy — APPROVED** [review]

**Comment by bleroy** [comment]
Thanks for the contribution.

**Pull request merged**